src/hooks/stores/useSyncStore.ts

```diff
@@ -0,0 +1,102 @@
+import { useMemo } from "react";
+import { create } from "zustand";
+
+type Progress = {
+  completed: number;
+  total: number;
+};
+
+type ActiveSync = {
+  progress: Progress;
+  syncIsFinished: boolean;
+};
+
+type AllSyncs = Record<string, ActiveSync>;
+
+type SyncStore = {
+  allSyncs: AllSyncs;
+  setIndivdualSync: (deviceId: string, progress?: Progress) => void;
+  resetSyncStore: () => void;
+};
+
+const emptyProgress: Progress = {
+  completed: 0,
+  total: 0,
+};
+
+const useSyncStore = create<SyncStore>()((set) => ({
```

Comment:
```suggestion
const useSyncStore = create<SyncStore>((set) => ({
```

Reply:
For [typescript](https://github.com/pmndrs/zustand/blob/main/docs/guides/typescript.md), you actually need that extra set of brackets